File: Core/src/main/java/jokatu/stomp/SendErrorMessage.java
package jokatu.stomp;

import ophelia.util.MapUtils;
import org.springframework.messaging.support.ErrorMessage;

/**
 * An error message returned to the client when one of its SEND commands is rejected.
 */
public class SendErrorMessage extends ErrorMessage {
	public SendErrorMessage(Throwable e, String receipt) {
		super(e, MapUtils.createMap("receipt-id", receipt));
	}
}

File: Core/src/main/kotlin/jokatu/stomp/SendErrorMessage.kt
package jokatu.stomp

import ophelia.util.MapUtils
import org.springframework.messaging.support.ErrorMessage

/**
 * An error message returned to the client when one of its SEND commands is rejected.
 */
class SendErrorMessage(e: Throwable, receipt: String) : ErrorMessage(e, MapUtils.createMap<String, Any>("receipt-id", receipt))
